src/plugins/mf_priority.cpp

```diff
@@ -198,6 +198,71 @@ int check_queue_factor (flux_plugin_t *p,
 }
 
 
+/*
+ * Add held job IDs to a JSON array to be added to a bank_info JSON object.
+ */
+static json_t *add_held_jobs (
+                            const std::pair<std::string, struct bank_info> &b)
+{
+    json_t *held_jobs, *jobid = NULL;
+
+    // add any held jobs to a JSON array
+    if (!(held_jobs = json_array ()))
+        goto error;
+
+    for (auto const &j : b.second.held_jobs) {
+        jobid = json_integer (j);
+        if (!jobid)
+            goto error;
+
+        if (json_array_append (held_jobs, jobid) < 0)
+            goto error;
+        
+        json_decref (jobid);
+    }
+
+    return held_jobs;
+error:
+    json_decref (held_jobs);
+    json_decref (jobid);
+    return NULL;
```

Comment:
This looks correct to me.

I do have one style suggestion: declare `jobid` within the for loop and instead of calling `json_decref (jobid)` at the `error:` label,  use
```c
if (json_array_append_new (held_jobs, jobid) < 0) :
    json_decref (jobid)
    goto error;
}
```
(and skip the `json_decref (jobid)` at the bottom of the for loop).

This is a simple case but I find it gets really confusing in general when you allocate something in a loop and free it outside of the loop.

Reply:
Ah, I like this style suggestion - thanks. This seems more straightforward to me to declare and free the variable within the for-loop, so I'll adopt this!